Added Detailed Recipe Page

marc0manuel opened the pull request:
Added Detailed Recipe Page. Note, the recipe page pulls from a data json array, when we want to go live, we will need to change this to a GET Request. Styling needs to be worked on. 

Added these files: 

IngredientList.jsx
InstructionsList.jsx
NutritionInfoCard.jsx
RecipeDetailPage.jsx
SimilarRecipes.jsx

marc0manuel committed:
Added Detailed Recipe Page

jerryiu424 reviewed (commented):
Update yo branch homie

marc0manuel committed:
Update branch, removed unecessary files

marc0manuel committed:
Merge branch 'master' into dev-manuel

jlee3227 commented on client/src/index.jsx:
```diff
@@ -16,7 +16,7 @@ ReactDOM.render(
     <Router>
       <Switch>
-        <Route path='/recipe/:recipeId'>
+        <Route path='/recipeId'>
```
Is this a typo? I can navigate to the Recipe Detail Page, but I think we want to use `recipeId`.

marc0manuel replied:
Not a typo, was working on an offline version so I dont make 10000 API calls in 10 seconds by accident. When we go to production we can change it all to be referencing the actual ID.

marc0manuel committed:
added serving size multiplier

marc0manuel committed:
Merge branch 'master' into dev-manuel

marc0manuel committed:
merging with master

marc0manuel committed:
resolved master issues

jerryiu424 committed:
Merge branch 'master' into dev-manuel

jerryiu424 committed:
Removes yarn.lock

maryamkarimi committed:
Merge remote-tracking branch 'origin/master' into dev-manuel

maryamkarimi committed:
Added checkboxes to ingredients (IngredientsCheck.jsx)

Moved the ingredients multiplier to the same row as "Ingredients" and IngredientsList.jsx (I think it makes more sense to have it there)

Re-aligned the ingredients and instructions

Removed unused imports and ran eslint to make indentation and spacing consistent

maryamkarimi committed:
Removed unused imports

maryamkarimi committed:
minor change

maryamkarimi committed:
Merge branch 'dev-manuel' of https://github.com/maryamkarimi/recipe-app into dev-manuel

maryamkarimi committed:
Added header to recipe detail page

LlemonDuck commented on client/src/index.jsx:
```diff
@@ -16,7 +16,7 @@ ReactDOM.render(
     <Router>
       <Switch>
-        <Route path='/recipe/:recipeId'>
+        <Route path='/recipeId'>
```
This change will eventually need to be reverted to allow for recipeId as a parameter, rather than the static path

LlemonDuck commented on client/src/pages/RecipeDetailPage.jsx:
```diff
@@ -1,7 +1,507 @@
+              <Typography variant="h5" gutterBottom>
+                Vegetarian:
+                {' '}
+                {data.vegetarian ? <SmileyIcon color="secondary" fontSize="medium" /> : <SadIcon color="primary" fontSize="medium" /> }
```
Can we use checkmarks and X's instead? The frowning face confused me for a while at first glance. Same for the below two.

LlemonDuck committed:
Use check and x marks for vege/gluten/keto instead of :) and :(
